docs/runbooks/VirtualMachineStuckOnNode.md

```diff
@@ -0,0 +1,265 @@
+   ```bash
+   kubectl delete pod <virt-launcher-pod> -n <namespace>
+   # The VMI controller will recreate it
+   ```
+
+2. **Check resource constraints**:
+   ```bash
+   kubectl describe pod <virt-launcher-pod> -n <namespace>
+   # Look for resource limit violations
+   ```
+
+### Image Issues on Node
+1. **Clear image cache** on the problematic node:
+   ```bash
+   # On the node (requires node access):
+   crictl rmi <problematic-image>
+   ```
+
+2. **Force image re-pull**:
+   ```bash
+   # Delete and recreate the virt-launcher pod
+   kubectl delete pod <virt-launcher-pod> -n <namespace>
+   ```
+
+### Storage Issues
+1. **Check PVC binding and mounting**:
+   ```bash
+   kubectl get pvc -n <namespace>
+   # If PVC is stuck, check the storage provisioner
+   ```
+
+2. **Resolve volume attachment issues**:
+   ```bash
+   kubectl get volumeattachment
+   # Delete stuck volume attachments if necessary
+   kubectl delete volumeattachment <attachment-name>
+   ```
+
+### Node-Level Issues  
+1. **Drain and uncordon node** if it's in a bad state:
+   ```bash
+   kubectl drain <node-name> --ignore-daemonsets --delete-emptydir-data
+   kubectl uncordon <node-name>
+   ```
+
+2. **Restart node-level components**:
+   ```bash
+   # Restart virt-handler on the node
+   kubectl delete pod <virt-handler-pod> -n kubevirt
+   
+   # If necessary, restart container runtime (requires node access)
+   systemctl restart containerd  # or crio
+   ```
+
+### VM-Level Resolution
+1. **Force VM restart**:
+   ```bash
+   # Stop and start the VM (will trigger new VMI creation)
+   virtctl stop <vm-name> -n <namespace>
+   virtctl start <vm-name> -n <namespace>
+   ```
+
+2. **Migrate VM to different node**:
+   ```bash
+   virtctl migrate <vm-name> -n <namespace>
+   ```
+
+### Emergency Actions
+- **Live migrate** critical VMs away from problematic node
+- **Force delete** stuck VMI if safe to do so:
+  ```bash
+  kubectl delete vmi <vm-name> -n <namespace> --force --grace-period=0
+  ```
+- **Cordon node** to prevent new VM scheduling while investigating
+
+## Prevention
+
+1. **Node Health Monitoring:**
+   - Monitor node resource utilization (CPU, memory, storage)
+   - Set up alerts for node conditions and taints
+   - Regular health checks on container runtime
+
+2. **Resource Management:**
+   - Set appropriate resource requests/limits on VMs
+   - Monitor PVC and storage utilization
+   - Plan for node capacity and VM density
+
+3. **Image Management:**
+   - Use image pull policies appropriately (Always, IfNotPresent)
+   - Pre-pull critical images to nodes
+   - Monitor image registry health and connectivity
+
+4. **Networking:**
+   - Ensure stable network connectivity between nodes and storage
+   - Monitor DNS resolution and service discovery
+   - Validate network policies don't block required traffic
+
+5. **Regular Maintenance:**
+   - Keep nodes and KubeVirt components updated
+   - Regularly restart long-running VMs to prevent resource leaks
```

Comment:
These are also rather unspecific suggestions, there should be no reason to kill long running VMs